tests/common/dnspolicy/dnspolicy_controller_test.go

```diff
@@ -1244,6 +1245,143 @@ var _ = Describe("DNSPolicy controller", func() {
+				g.Expect(recordList.Items).To(HaveLen(2))
+
+				// This record should not be ready - we are not publishing unhealthy EPs
+				dnsRecord := &kuadrantdnsv1alpha1.DNSRecord{}
+				g.Expect(k8sClient.Get(ctx, client.ObjectKey{Name: recordName, Namespace: testNamespace}, dnsRecord)).To(Succeed())
+				g.Expect(dnsRecord.Status.Conditions).To(
+					ContainElements(
+						MatchFields(IgnoreExtras, Fields{
+							"Type":    Equal(string(kuadrantdnsv1alpha1.ConditionTypeReady)),
+							"Status":  Equal(metav1.ConditionFalse),
+							"Reason":  Equal(string(kuadrantdnsv1alpha1.ConditionReasonUnhealthy)),
+							"Message": Equal("Not publishing unhealthy records"),
+						}),
+						MatchFields(IgnoreExtras, Fields{
+							"Type":   Equal(string(kuadrantdnsv1alpha1.ConditionTypeHealthy)),
+							"Status": Equal(metav1.ConditionFalse),
+							"Reason": Equal(string(kuadrantdnsv1alpha1.ConditionReasonUnhealthy)),
+							"Message": And(
+								ContainSubstring("Not healthy addresses"),
+								ContainSubstring(tests.IPAddressOne),
+								ContainSubstring(tests.IPAddressTwo)),
+						}),
+					),
+				)
+
+				// This record should be ready - we are not creating checks for wildcards
+				wildcardDnsRecord := &kuadrantdnsv1alpha1.DNSRecord{}
+				g.Expect(k8sClient.Get(ctx, client.ObjectKey{Name: wildcardRecordName, Namespace: testNamespace}, wildcardDnsRecord)).To(Succeed())
+				g.Expect(wildcardDnsRecord.Status.Conditions).To(
+					And(
+						ContainElement(
+							MatchFields(IgnoreExtras, Fields{
+								"Type":    Equal(string(kuadrantdnsv1alpha1.ConditionTypeReady)),
+								"Status":  Equal(metav1.ConditionTrue),
+								"Reason":  Equal(string(kuadrantdnsv1alpha1.ConditionReasonProviderSuccess)),
+								"Message": Equal("Provider ensured the dns record"),
+							}),
+						),
+						Not(ContainElement(
+							MatchFields(IgnoreExtras, Fields{
+								"Type": Equal(string(kuadrantdnsv1alpha1.ConditionTypeHealthy)),
+							}),
+						)),
+					),
+				)
+
+				//Check policy status
+				err = k8sClient.Get(ctx, client.ObjectKeyFromObject(dnsPolicy), dnsPolicy)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(dnsPolicy.Status.Conditions).To(
+					ContainElements(
+						MatchFields(IgnoreExtras, Fields{
+							"Type":    Equal(string(gatewayapiv1alpha2.PolicyConditionAccepted)),
+							"Status":  Equal(metav1.ConditionTrue),
+							"Reason":  Equal(string(gatewayapiv1alpha2.PolicyConditionAccepted)),
+							"Message": Equal("DNSPolicy has been accepted"),
+						}),
+						MatchFields(IgnoreExtras, Fields{
+							"Type":   Equal(string(kuadrant.PolicyConditionEnforced)),
+							"Status": Equal(metav1.ConditionTrue),
+							"Reason": Equal(string(kuadrant.PolicyReasonEnforced)),
+							"Message": And(
+								ContainSubstring("DNSPolicy has been partially enforced. Not ready DNSRecords are:"),
+								ContainSubstring(recordName),
+								Not(ContainSubstring(wildcardRecordName))),
```

Comment:
If we are partially enforced with only two records it is sufficient to only check for the presence of one of the records. Leaving it here to be extra cautious.